pyleros/alu.py

```diff
@@ -0,0 +1,72 @@
+from myhdl import instances, block, intbv, \
+                    always_comb, always_seq
+
+from pyleros.types import alu_op_type, t_decSignal
+
+
+
+@block
+def pyleros_alu(dec, acc, opd, pre_acc):
+    """The alu module for pyleros, purely combinatorial
+
+    Arguments (ports):
+        dec: IN The decoder control signals
+        acc: IN The accumulator
+        opd: IN Operand, based on whether the instruction is 
+            immediate or not
+        pre_acc: OUT The result of the ALU operation, from the acc mux
+
+    Parameters:
+        None
+    """
+
+    res_arith, res_log = intbv(0)[16:], intbv(0)[16:]
+
+    # Add and Subtract module
+    @always_comb
+    def op_add_sub():
+
+        if not dec[int(t_decSignal.add_sub)]:
```

Comment:
I think you are making this more complicated than it needs, instead of creating the list-of-signals and trying to map a type to the particular element in the list-of-signals, I think you simply need to:

```python 
dec = Signal(dec_op_type.op)   # some default op

alu_inst = alu(dec, acc, opd, pre_acc)
```

```python
def alu(dec, acc, opd, pre_acc):
    if dec == dec_op_type.add_sub:
        res_arith = (acc - opd) & 0xFFFF
    else:
        res_arith = (acc + opd) & 0xFFFF
```